src/Chip/Chip.js

```diff
@@ -0,0 +1,270 @@
+import React, {Component, PropTypes} from 'react';
+import keycode from 'keycode';
+import {emphasize} from '../utils/colorManipulator';
+import EnhancedButton from '../internal/EnhancedButton';
+import DeleteIcon from '../svg-icons/navigation/cancel';
+
+function getStyles(props, context, state) {
+  const {chip} = context.muiTheme;
+
+  const pressedColor = props.backgroundColor ? emphasize(props.backgroundColor, 0.1) : chip.pressedColor;
+  const backgroundColor = props.backgroundColor || chip.backgroundColor;
+
+  return {
+    avatar: {
+      marginRight: -4,
+    },
+    closeIcon: {
+      color: chip.closeIconColor,
+      margin: '4px 4px 0px -8px',
+    },
+    label: {
+      color: props.labelColor || chip.textColor,
+      fontSize: 13,
+      fontWeight: chip.fontWeight,
+      lineHeight: '32px',
+      paddingLeft: 12,
+      paddingRight: 12,
+      whiteSpace: 'nowrap',
+    },
+    root: {
+      backgroundColor: state.clicked ? pressedColor : backgroundColor,
+      borderRadius: 16,
+      boxShadow: state.focused ? chip.shadow : null,
+      display: 'flex',
+      whiteSpace: 'nowrap',
+      width: 'fit-content',
+    },
+  };
+}
+
+class Chip extends Component {
+
+  static propTypes = {
+
+    /**
+     * Override the background color of the chip.
+     */
+    backgroundColor: PropTypes.string,
+
+    /**
+     * Used to render elements inside the Chip.
+     */
+    children: PropTypes.node,
+
+    /**
+     * CSS `className` of the root element.
+     */
+    className: PropTypes.node,
+
+    /**
+     * Override the label color.
+     */
+    labelColor: PropTypes.string,
+
+    /**
+     * Override the inline-styles of the label.
+     */
+    labelStyle: PropTypes.object,
+
+    /** @ignore */
+    onBlur: PropTypes.func,
+
+    /** @ignore */
+    onFocus: PropTypes.func,
+
+    /** @ignore */
+    onKeyDown: PropTypes.func,
+
+    /** @ignore */
+    onKeyboardFocus: PropTypes.func,
+
+    /** @ignore */
+    onMouseDown: PropTypes.func,
+
+    /** @ignore */
+    onMouseEnter: PropTypes.func,
+
+    /** @ignore */
+    onMouseLeave: PropTypes.func,
+
+    /** @ignore */
+    onMouseUp: PropTypes.func,
+
+    /**
+     * Callback function fired when the delete icon is clicked. If set, the delete icon will be shown.
+     * @param {object} event `touchTap` event targeting the element.
+     */
+    onRequestDelete: PropTypes.func,
+
+    /** @ignore */
+    onTouchEnd: PropTypes.func,
+
+    /** @ignore */
+    onTouchStart: PropTypes.func,
+
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: PropTypes.object,
+  };
+
+  static defaultProps = {
+    onBlur: () => {},
```

Comment:
@mbrookes in another PR @oliviertassinari ended up going with `typeof` checks instead of creating a noop function. Should we have a consistent implementation throughout the lib? I'll let you guys hash it out 😄